lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -18,47 +19,71 @@ def exists?
   def create
     Puppet.debug "Creating #{@resource[:name]}"
 
-    if @resource[:physicalpath].nil?
-      fail "physicalpath is a required paramter to create a iis virtual directory"
-    end
+    verify_physicalpath
 
-    if @resource[:physicalpath] and ! File.exists?(@resource[:physicalpath])
-      fail "physicalpath doesn't exist: #{@resource[:physicalpath]}"
-    end
+    # New-WebVirtualDirectory fails when PhysicalPath is a UNC path that is not available:
+    #   "Parameter 'PhysicalPath' should point to existing path."
 
-    cmd = []
-    cmd << "New-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
-    cmd << "-Site \"#{@resource[:sitename]}\" " if @resource[:sitename]
-    cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
-    cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" " if @resource[:physicalpath]
-    cmd << "-ErrorAction Stop"
-    cmd = cmd.join
+    if is_drive_path(@resource[:physicalpath])
+      cmd = []
+      cmd << "New-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
+      cmd << "-Site \"#{@resource[:sitename]}\" " if @resource[:sitename]
+      cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
+      cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" "
+      cmd << "-ErrorAction Stop"
+      cmd = cmd.join
 
-    result   = self.class.run(cmd)
-    Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
-    @resource[:ensure] = :present
+      result   = self.class.run(cmd)
+      Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
+      @resource[:ensure] = :present
+    end
+    if is_unc_path(@resource[:physicalpath])
+      cmd = []
+      cmd << "New-Item -type VirtualDirectory \"IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}\" "
+      cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
+      cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" "
+      cmd = cmd.join
+
+      result = self.class.run(cmd)
+      Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
+      @resource[:ensure] = :present
+    end
   end
 
   def update
+    # WORKAROUND: update() is called after destroy() by flush() in ../iis_powershell.rb
+    # New-Item -Force recreates the VirtualDirectory unless we return early.
+    return if (@resource[:ensure] == :absent)
```

Comment:
I'm not sure this is necessary, what is described in the comments is how Puppet works. Can you describe the reason to abort early here?

Reply:
The update method recreates the virtual directory using 'Set-ItemProperty' after the destroy method removes the virtual directory using Remove-WebVirtualDirectory. Maybe destroy() should not call update() instead?

Reply:
After your rebase we'll need to test this to make sure its working as expected